projects/metis/src/app/dataset/history/history.component.html

```diff
@@ -6,42 +6,52 @@
 ></app-notification>
 
 <div class="table-grid history">
-  <div class="table-grid-header">
-    {{ 'workflowStep' | translate }}
-  </div>
-  <div class="table-grid-header">
-    {{ 'processedRecords' | translate }}
-  </div>
-  <div class="table-grid-header date-header">
-    {{ 'startTime' | translate }}
-  </div>
-  <div class="table-grid-header date-header">
-    {{ 'endTime' | translate }}
-  </div>
-  <div class="table-grid-header"></div>
-
-  <ng-container *ngFor="let wpe of allExecutions; trackBy: byId">
-    <div
-      *ngIf="!wpe.pluginExecution && wpe.execution; let execution"
-      class="sub-header workflow-head"
-    >
+  <ng-container *ngFor="let wpe of allExecutions; trackBy: byId; let i = index">
```

Comment:
I think this container and everything in it is just unnecessarily complicated. You have this strange structure called allExecutions that tries to flatten the hierarchy of workflow executions and plugin executions inside them. I don't see the need to do that.

Why not have a list of workflow executions, and then you have the first ng-container below, and then you loop through the plugins in the workflow to have instances of `app-executions-grid-data`? That way the code makes much more sense, you don't need an awkward data structure and, even better, you can then compute the index as you normally would by taking the modulo, instead of having to use this awkward `rowIsStripe` function.